solr/core/src/java/org/apache/solr/handler/component/ResponseBuilder.java

```diff
@@ -175,6 +175,23 @@ public void addRequest(SearchComponent me, ShardRequest sreq) {
     }
   }
 
+  /**
+   *  Override this method and the {@link ResponseBuilder#addShardInfo(Object, String, NamedList)}
+   *  method e.g. if you wish to return a list instead of a map container.
+   */
+  protected Object newShardsInfo() {
```

Comment:
Should this return some interface instead of an Object?

Reply:
> Should this return some interface instead of an Object?

Yes, that's a good idea, thank you. It would also avoid the _"if you changed this method you must also change the other method"_ stuff.